nifi-stateless/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/flow/StandardStatelessFlow.java

```diff
@@ -521,6 +522,26 @@ private void executeDataflow(final BlockingQueue<TriggerResult> resultQueue, fin
         }
     }
 
+    @Override
+    public boolean isStateful() {
+        return isStateful(rootGroup);
```

Comment:
Not sure how frequently this will be called. But since the value won't change from call to call, wondering if it makes sense to lazily initialize this and just cache the value in a member variable?